beacon-chain/powchain/deposit.go

```diff
@@ -12,6 +13,10 @@ func (s *Service) processDeposit(ctx context.Context, eth1Data *ethpb.Eth1Data,
 	if err := s.preGenesisState.SetEth1Data(eth1Data); err != nil {
 		return err
 	}
-	s.preGenesisState, err = blocks.ProcessPreGenesisDeposits(ctx, s.preGenesisState, []*ethpb.Deposit{deposit})
-	return err
+	beaconState, err := blocks.ProcessPreGenesisDeposits(ctx, s.preGenesisState, []*ethpb.Deposit{deposit})
+	if err != nil {
+		return errors.Wrap(err, "could not process pre-genesis deposits")
+	}
+	s.preGenesisState = beaconState
```

Comment:
want to check that beaconState is not nil? Seems unlikely but it would be the defensive thing to do, i suppose